stdlib/REPL/src/LineEdit.jl

```diff
@@ -395,16 +397,24 @@ function refresh_multi_line(termbuf::TerminalBuffer, terminal::UnixTerminal, buf
     # Now go through the buffer line by line
     seek(buf, 0)
     moreinput = true # add a blank line if there is a trailing newline on the last line
+    lastline = false # indicates when to stop printing lines, even when there are potentially
+                     # more (for the case where rows is too small to print everything)
+                     # Note: when there are too many lines for rows, we still print the first lines
+                     # even if they are going to not be visible in the end: for simplicity, but
+                     # also because it does the 'right thing' when the window is resized
     while moreinput
-        l = readline(buf, keep=true)
-        moreinput = endswith(l, "\n")
+        line = readline(buf, keep=true)
+        moreinput = endswith(line, "\n")
+        if moreinput && lastline # we want to print only one "visual" line, so
+            line = line[1:end-1] # don't include the trailing "\n"
```

Comment:
use chomp (so that this won't fail if the last character is multi-byte)

Reply:
Ah right, maybe `prevind(line, end-1)` is fine too, as we know the last character is `'\n'` ?

Reply:
we know that `end` is `\n`, but that tells nothing about `end-1` (or `prevind(line, endof(line))`)—but `chomp`/`chop` is simpler